polaris-core/src/main/java/org/apache/polaris/core/persistence/MetaStoreManagerFactory.java

```diff
@@ -34,9 +35,11 @@ public interface MetaStoreManagerFactory {
 
   Supplier<? extends BasePersistence> getOrCreateSessionSupplier(RealmContext realmContext);
 
-  StorageCredentialCache getOrCreateStorageCredentialCache(RealmContext realmContext);
+  StorageCredentialCache getOrCreateStorageCredentialCache(
+      RealmContext realmContext, PolarisCallContext polarisCallContext);
 
-  EntityCache getOrCreateEntityCache(RealmContext realmContext);
+  EntityCache getOrCreateEntityCache(
+      RealmContext realmContext, PolarisCallContext polarisCallContext);
```

Comment:
I do not support passing context parameter as method args in general and this PR adds another context parameter.

Could we leverage CDI for injecting context data?

Reply:
This is a good callout. In general I agree that we can leverage CDI to inject context parameters (e.g. PolarisCallContext) around. However in this particular case, the EntityCache is not a CDI-managed class. There's a discussion above about this too, but I think we need a way to have configs that are not tied to a PolarisCallContext('s configuration store).